src/FastField.tsx

```diff
@@ -110,20 +110,22 @@ class FastFieldInner<Values = {}, Props = {}> extends React.Component<
   ) {
     if (this.props.shouldUpdate) {
       return this.props.shouldUpdate(props, this.props);
-    } else if (
-      getIn(this.props.formik.values, this.props.name) !==
-        getIn(props.formik.values, this.props.name) ||
-      getIn(this.props.formik.errors, this.props.name) !==
-        getIn(props.formik.errors, this.props.name) ||
-      getIn(this.props.formik.touched, this.props.name) !==
-        getIn(props.formik.touched, this.props.name) ||
-      Object.keys(this.props).length !== Object.keys(props).length ||
-      this.props.formik.isSubmitting !== props.formik.isSubmitting
-    ) {
-      return true;
-    } else {
-      return false;
     }
+    if (props.formik.lastTouched === this.props.name) {
```

Comment:
Why not put this all in a single return statement, something like this?

```js
    return (props.formik.lastTouched === this.props.name) && (
        this.props.formik.isSubmitting !== props.formik.isSubmitting ||
        Object.keys(this.props).length !== Object.keys(props).length ||
        getIn(this.props.formik.values, this.props.name) !==
          getIn(props.formik.values, this.props.name) ||
        getIn(this.props.formik.errors, this.props.name) !==
          getIn(props.formik.errors, this.props.name) ||
        getIn(this.props.formik.touched, this.props.name) !==
          getIn(props.formik.touched, this.props.name)
      );
```

Reply:
Or more drastically rewrite the function body like this:

```js
    return this.props.shouldUpdate) ?
      this.props.shouldUpdate(props, this.props) :
      (props.formik.lastTouched === this.props.name) && (
        this.props.formik.isSubmitting !== props.formik.isSubmitting ||
        Object.keys(this.props).length !== Object.keys(props).length ||
        getIn(this.props.formik.values, this.props.name) !==
          getIn(props.formik.values, this.props.name) ||
        getIn(this.props.formik.errors, this.props.name) !==
          getIn(props.formik.errors, this.props.name) ||
        getIn(this.props.formik.touched, this.props.name) !==
          getIn(props.formik.touched, this.props.name)
      );
```

Reply:
i hardly prefer your first suggestion for reading purpose.

note: is it a fast review because it relies on fast fields ? :p

Reply:
changed it